labonneboite/web/admin/views/office_admin_update.py

```diff
@@ -298,6 +299,38 @@ class OfficeAdminUpdateModelView(AdminModelViewMixin, ModelView):
         },
     }
 
+    def _handle_view(self, name, **kwargs):
+        # This method is executed before calling any view method.
+
+        """
+            Sometimes, recruiter send two emails at the same time. Both conatins a create office_admin_update link.
+            But, when the first office_admin_update is created, the suggestion in the second mail became wrong.
+            And, on saving, we got a conflict with the recently created office_admin_update.
+            To avoid that, we redirect to the edit page if, afterall, an office_admin_update already exists for the company siret.
+        """
+        if name == 'create_view':
```

Comment:
Ici on surcharge une méthode privée (dont le nom commence par "_") alors qu'on ne fait réellement quelque chose que lorsque `name == 'create_view'`. Est-ce que ça ne serait pas plus simple de surcharger la méthode `create_view` ? 

```
def create_view(self, *args, **kwargs):
    recruiter_message_params = self.get_recruiter_message_params()
    ...
    return super().create_view(*args, **kwargs)
```

Reply:
Mes essais avec `create_view` n'ont rien donné...

D'ailleurs, sur le quickstart de flask-admin, la méthode n'a pas l'air si privée que ça vu qu'on indique de la modifier si besoin...
https://github.com/mrjoes/flask-admin/blob/master/doc/quickstart.rst#authentication

Reply:
comment ça "rien donné" ? on ne rentre pas dans le code de la méthode ?

Reply:
On y rentre pas et `http://localhost:5000/admin/officeadminupdate/` renvoie une erreur `werkzeug.routing.BuildError: Could not build url for endpoint 'officeadminupdate.create_view' with values ['url']. Did you mean 'officeadminupdate.delete_view' instead ?`